src/main/java/org/codehaus/mojo/license/api/DefaultThirdPartyTool.java

```diff
@@ -754,59 +750,80 @@ private void loadOneGlobalSet( SortedSet<MavenProject> unsafeDependencies, Licen
-                    project.getGroupId(), project.getArtifactId(), project.getVersion(),
-                    DESCRIPTOR_TYPE, DESCRIPTOR_CLASSIFIER )
-                    : e.getArtifact();
+                org.eclipse.aether.artifact.Artifact artifact;
+                if ( artifactNotFoundException.getArtifact() == null )
+                {
+                    artifact = new DefaultArtifact( project.getGroupId(), project.getArtifactId(),
+                            DESCRIPTOR_CLASSIFIER, null, project.getVersion(),
+                            new DefaultArtifactType( DESCRIPTOR_TYPE ) );
+                }
+                else
+                {
+                    artifact = artifactNotFoundException.getArtifact();
+                }
 
-            // we can afford to write an empty descriptor here as we don't expect it to turn up later in the remote
-            // repository, because the parent was already released (and snapshots are updated automatically if changed)
-            result = new File( localRepository.getBasedir(), localRepository.pathOf( artifact ) );
+                /*
+                 * we can afford to write an empty descriptor here
+                 * as we don't expect it to turn up later in the remote
+                 * repository, because the parent was already released
+                 * (and snapshots are updated automatically if changed)
+                 */
+                result = new File( aetherRepoSession.getLocalRepository().getBasedir(),
+                        aetherRepoSession.getLocalRepositoryManager().getPathForLocalArtifact( artifact ) );
+            }
+            else
+            {
+                throw e;
+            }
         }
 
         return result;
     }
 
     public File resolveMissingLicensesDescriptor( String groupId, String artifactId, String version,
-            ArtifactRepository localRepository, List<ArtifactRepository> repositories )
+            List<RemoteRepository> remoteRepositories )
             throws IOException, ArtifactResolutionException, ArtifactNotFoundException
     {
-        return resolveArtifact( groupId, artifactId, version, DESCRIPTOR_TYPE, DESCRIPTOR_CLASSIFIER, localRepository,
-                repositories );
+        return resolveArtifact( groupId, artifactId, version, DESCRIPTOR_TYPE,
+                DESCRIPTOR_CLASSIFIER, remoteRepositories );
     }
 
     private File resolveArtifact( String groupId, String artifactId, String version,
-            String type, String classifier, ArtifactRepository localRepository, List<ArtifactRepository> repositories )
-                    throws ArtifactResolutionException, IOException, ArtifactNotFoundException
+            String type, String classifier, List<RemoteRepository> remoteRepositories )
+                    throws ArtifactResolutionException
     {
         // TODO: this is a bit crude - proper type, or proper handling as metadata rather than an artifact in 2.1?
```

Comment:
Shouldn't this comment be removed? It does not seem to make any sense anymore.

Reply:
Maybe? I understood the comment to say that handling the third party descriptor as an artifact was crude, but I'm not sure what the intended alternative was. I'll remove the comment.

Reply:
Removed the comment